File: 07/7.rb
def main()
    i = 1
    count = 0
    while (count != 10001)
        if (isPrime(i))
            count += 1
        end
        i += 2
    end
    puts i-2
end 

def isPrime(n)
    if (n < 4)
        return true
    end
    if (n % 2 == 0)
        return false
    end
    for i in (3..n**0.5).step(2)
        if (n % i == 0)
            return false
        end
    end
    return true
end

if __FILE__ == $0
    main()
end

File: 07/7.py
def main():
     i = 1
     count = 0
     while (count != 10001):
         if (isPrime(i)):
             count = count + 1
         i = i + 2
     print (i - 2)

def isPrime(n):
    if (n < 4):
        return True
    if (n % 2 == 0):
        return False
    for i in range(3 , int((n**0.5) + 1)  , 2):
        if (n % i == 0):
            return False
    return True

main()
